clang-tools-extra/clang-tidy/bsl/DestructorAccessSpecifierCheck.cpp

```diff
@@ -0,0 +1,53 @@
+//===--- DestructorAccessSpecifierCheck.cpp - clang-tidy ------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "DestructorAccessSpecifierCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace bsl {
+
+void DestructorAccessSpecifierCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(cxxDestructorDecl( 
+                    unless(anyOf(cxxDestructorDecl(isPublic()),
+                           cxxDestructorDecl(isPublic(), isOverride()),
```

Comment:
The inner cxxDestructorDecl's aren't required

Reply:
cppreference does say "In a member function declaration or definition, override ensures that the function is virtual and is overriding a virtual function from a base class," but I'm a little confused since my understanding is that "override" is complementary to "virtual," so you would mark the base class function as virtual and the subclass function as override. tldr; I guess checking for public virtual encompasses public override and then a little more, but that's ok?